src/spl/parameter_space.h

```diff
@@ -223,6 +202,23 @@ class ParameterSpace {
     }
   }
 
+  void RecreateBasisFunctions() {
+    for (int current_dim = 0; current_dim < DIM; ++current_dim) {
+      auto &basis_functions_current_dim = basis_functions_[current_dim];
+      auto &knot_vector_current_dim = knot_vector_[current_dim];
+      auto &degree_current_dim = degree_[current_dim];
+      basis_functions_current_dim.erase(basis_functions_current_dim.begin(), basis_functions_current_dim.end());
+      basis_functions_current_dim.reserve(knot_vector_current_dim->GetNumberOfKnots() - degree_current_dim.get() - 1);
+      for (int current_basis_function = 0;
+           current_basis_function <
+               (static_cast<int>(knot_vector_current_dim->GetNumberOfKnots()) - degree_current_dim.get() - 1);
```

Comment:
Can you remove the static cast? Could you do the line break after the minus operator?